torch/library.py

```diff
@@ -49,9 +49,43 @@ def fallthrough_kernel():
     raise NotImplementedError("fallthrough_kernel() should never be called.")
 
 
-def Schema(name: str, function: Callable, mutates_args=()) -> str:
+def infer_schema(
+    function: Callable, name: Optional[str] = None, mutates_args=()
+) -> str:
+    r"""Parses the schema of a given function with type hints. The schema is inferred from the
+        function's type hints, and can be used to define a new operator.
+
+    We make the following assumptions:
+    - None of the outputs alias any of the inputs or each other.
+    - String type annotations "device, dtype, Tensor, types" without library specification
+      are assumed to be torch.*. Similarly, string type annotations "Optional, List, Sequence, Union"
+      without library specification are assumed to be typing.*.
+    - Only the args listed in ``mutates_args`` are being mutated. If ``mutates_args`` is "unknown",
+        it assumes that all inputs to the operator are being mutates.
+
+    Callers (e.g. the custom ops API) are responsible for checking these assumptions.
+
+    Args:
+        function: function schema to define a new operator.
+        name (Optional[str]): The name of the operator in the schema. If ``name`` is None, then the
+                              name of the function is used as the name of the operator.
+        mutates_args ("unknown" | Iterable[str]): The arguments that are mutated in the function.
+
+    Returns:
+        The inferred schema.
+
+    Example::
+
+        >>> def foo_impl(x: torch.Tensor) -> torch.Tensor:
+        >>>     return x.sin()
+        >>>
+        >>> torch.library.infer_schema(function=foo_impl, name="foo", mutates_args={})
+        >>> # foo(Tensor x) -> Tensor
+    """
     from torch._library.infer_schema import infer_schema
 
+    if name is None:
+        name = function.__name__
```

Comment:
Can we beef up torch._library.infer_schema take in `name` and instead just `from torch._library.infer_schema import infer_schema` at the top of this file? that would mean that we don't need two `infer_schema` functions

Reply:
changed.